enhancements/network/ovn-kubernetes-multiple-external-gateway.md

```diff
@@ -0,0 +1,177 @@
+
+The annotation logic will be removed in OpenShift 4.15.
+
+### Workflow Description
+The workflow follows the behavior of a cluster scoped CR.
+
+
+### API Extensions
+
+A new cluster scoped CRD `AdminPolicyBasedExternalRoute` is created to represent the multiple external gateways configuration. The `spec` structure exposes the following fields:
+
+* `from`: determines where the routing polices should be applied to. Only `namespaceSelector` is supported for external traffic. This field is mandatory.
+* `nextHops`: defines the destination where the packets should be forwarded to. There are two types of references: static and dynamic. Static IPs usually refer to interfaces that hold an IP which doesn't change overtime Dynamic IPs link to pod IPs and can potentially change over time as they are linked to the pod's lifecycle.
+  *`static`: is a slice of static IPs, each item containing the following fields:
+      * `ip`: IPv4 or IPv6 of the next destination hop. This field is mandatory.
+      * `skipHostSNAT`: When true, it disables applying the Source NAT to the host IP. This field is optional and defaults to false.
+      * `bfdEnabled`: determines if Bi-Directional Forwarding Detection (BFD) is supported in this network and follows the same behavior as with the annotation: it's enabled when set to true and disabled when false or omited in the manifest. It maps to the `k8s.ovn.org/bfd-enabled` annotation in the pod. This field is optional and defaults to false.
+  * `dynamic`: references contain the following fields:
+    * `podSelector` implements a [set-based](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement) label selector to filter the pods in the namespace that match this network configuration. These are dynamic IPs linked to the lifecycle of the pod. This field is mandatory.
+    * `namespaceSelector` It defines a `set-based` selector that is used to filter the namespaces where to apply the `podSelector`. This field is optional.
+    * `skipHostSNAT`: When true, it disables applying the Source NAT to the host IP. This field is optional and defaults to false.
+    * `bfdEnabled`: determines if Bi-Directional Forwarding Detection (BFD) is supported in this network and follows the same behavior as with the annotation: it's enabled when set to true and disabled when false or omited in the manifest. It maps to the `k8s.ovn.org/bfd-enabled` annotation in the pod. This field is optional and defaults to false.
+    * `networkAttachmentName`: Name of the network attached definition. The name needs to match from the list of logical networks generated by Multus and associated to the pod via annotation. This field is optional. When this field is empty or not defined, the IP from the pod's host network will be used instead. The pod must have `hostnetwork` enabled in order for this feature to work.
+
+The following yaml examples an instance of the CRD:
+
+```yaml
```

Comment:
TODO: As discussed if any changes come in the API based on implementation, we can come back to update the fields, so loosely lgtm-ing on this initial one.